packages/@aws-cdk/aws-kinesisfirehose-destinations/lib/private/helpers.ts

```diff
@@ -91,3 +91,39 @@ export function createEncryptionConfig(role: iam.IRole, encryptionKey?: kms.IKey
     ? { kmsEncryptionConfig: { awskmsKeyArn: encryptionKey.keyArn } }
     : { noEncryptionConfig: 'NoEncryption' };
 }
+
+export function createProcessingConfig(
+  scope: Construct,
+  role: iam.IRole,
+  dataProcessors?: firehose.IDataProcessor[],
+): firehose.CfnDeliveryStream.ProcessingConfigurationProperty | undefined {
+  if (dataProcessors && dataProcessors.length > 1) {
+    throw new Error('Only one processor is allowed per delivery stream destination');
+  }
```

Comment:
Do we know what's the history of this limitation, and if there are any plans to lift it in the future?

Reply:
We are not aware of any plans to lift it in the future.

Reply:
Can we not make this an array then? I don't like this level of future-proofing if there are not concrete plans to lift this limitation.

If it is lifted in the far future, we have techniques to deal with that in the CDK.